### PR TITLE
[MM-44842] Add restore_group permission

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

This PR introduces a new permission to restore custom user groups, this action until now relied on permission to delete user groups.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-44842
#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by cyrilzhang-mm — APPROVED** [review]

**Review by BenCookie95 — APPROVED** [review]
Do we need any web app changes to surface this in the system console UI?

File: app/permissions_migrations.go [hunk 1]
```diff
@@ -979,6 +979,7 @@ func (a *App) getAddCustomUserGroupsPermissions() (permissionsMap, error) {
 		model.PermissionManageCustomGroupMembers.Id,
 		model.PermissionEditCustomGroup.Id,
 		model.PermissionDeleteCustomGroup.Id,
+		model.PermissionRestoreCustomGroup.Id,
```
[the remark this answers is not in the document]
Comment (plant99) [reply app/permissions_migrations.go:982]: Thanks Martin, working on this.

[the remark this answers is not in the document]
Comment (plant99) [reply app/permissions_migrations.go:1001]: Done, please review.

**Comment by plant99** [comment]
> Do we need any web app changes to surface this in the system console UI?

Thanks for raising this @BenCookie95  I don't think so though, groups can potentially be restored only through [mmctl](https://github.com/mattermost/mmctl/pull/614) for now.

**Comment by plant99** [comment]
> Do we need any web app changes to surface this in the system console UI?

@BenCookie95 https://github.com/mattermost/mattermost-webapp/pull/11760, thanks!

**Review by furqanmlk — APPROVED** [review]
Working fine

**Without permission**
image

**Permission**
image

**Comment by mkraft** [comment]
@plant99 is this ready to merge?

**Pull request merged**